docs/source/debugging.rst

```diff
@@ -0,0 +1,196 @@
+
+.. code-block:: python
+
+   >>> x.compute()
+   ZeroDivisionError(...)
+
+   >>> %pdb
+   >>> client.recreate_error_locally(x)
+
+
+Inspect Scheduling State
+------------------------
+
+Not all errors present themselves as Exceptions.  For example in a distributed
+systems workers may die unexpectedly or your computation may be unreasonably
+slow due to inter-worker communication or scheduler overhead or one of several
+other issues.  Getting feedback about what's going on can help to identify
+both failures and general performance bottlenecks.
+
+For the single-machine scheduler see :doc:`diagnostics <diagnostics`
+documentation.  The rest of the section will assume that you are using the
+`distributed scheduler <http://distributed.readthedocs.io/en/latest/>`_ where
+these issues arise more commonly.
+
+Web Diagnostics
+~~~~~~~~~~~~~~~
+
+First, the distributed scheduler has a number of diagnostic web pages showing
+dozens of recorded metrics like CPU, memory, network, and disk use, a history
+of previous tasks, allocation of tasks to workers, worker memory pressure, work
+stealing, open file handle limits, etc..  *Many* problems can be correctly
+diagnosed by inspecting these pages.  By default these are available at
+``http://scheduler:8787/`` ``http://scheduler:8788`` and ``http://worker:8789``
+where ``scheduler`` and ``worker`` should be replaced by the addresses of the
+scheduler and each of the workers.
+
+Logs
+~~~~
+
+The scheduler and workers and client all emits logs using `Python's standard
+logging module <https://docs.python.org/3/library/logging.html>`_.  By default
+these emit to standard error.  When Dask is launched by a cluster job scheduler
+(SGE/SLURM/YARN/Mesos/Marathon/Kubernetes/whatever) that system will track
+these logs and will have an interface to help you access them.  If you are
+launching Dask on your own they will probably dump to the screen unless you
+`redirect stderr to a file
+<https://en.wikipedia.org/wiki/Redirection_(computing)#Redirecting_to_and_from_the_standard_file_handles>`_
+.
+
+You can control the logging verbosity in the ``~/.dask/config.yaml`` file.
+Defaults currently look like the following:
+
+.. code-block:: yaml
+
+   logging:
+     distributed: info
+     distributed.client: warning
+     bokeh: error
+
+So for example you could add a line like ``distributed.worker: debug`` to get
+*very* verbose output from the workers.
+
+
+LocalCluster
+------------
+
+If you are using the distributed scheduler from a single machine you may be
+setting up workers manually using the command line interface or you may be
+using `LocalCluster <http://distributed.readthedocs.io/en/latest/local-cluster.html>`_
+which is what runs when you just call ``Client()``
+
+.. code-block:: python
+
+   >>> from dask.distributed import Client, LocalCluster
+   >>> client = Client()  # This is actually the following two commands
+
+   >>> cluster = LocalCluster()
+   >>> client = Client(cluster.scheduler.address)
```

Comment:
The client does have a reference at `client.cluster`, but when debugging I find it's nice to have the cluster constructor exposed.  That way you can control keywords on it.  In these docs it also makes it more explicit that there is another object in play.